core/server/data/migrations/versions/4.12/02-fix-member-statuses.js

```diff
@@ -0,0 +1,27 @@
+const logging = require('@tryghost/logging');
+const {createTransactionalMigration} = require('../../utils.js');
+
+module.exports = createTransactionalMigration(
+    async function up(knex) {
+        logging.info('Updating members with no products to "free"');
+
+        const freeMembers = await knex('members')
+            .select('members.id')
+            .leftJoin('members_products', 'members.id', 'members_products.member_id')
+            .whereNull('members_products.member_id');
```

Comment:
Should we also add a `.where('members.status', '!=', 'free')` to avoid doing extra work?

Reply:
Yeah, that makes sense!

Reply:
Is the `member_product` deleted in this case, as in is it `NULL` ? After going through reproduction steps in https://github.com/TryGhost/Team/issues/959 the `members_products` record still exists for me after cancelling the subscription from Admin UI :thinking:

Reply:
@naz Do you have webhooks set up locally to handle the cancelled subscription request? The product should be removed from the member, so the `members_products` entry will be deleted

Reply:
Verified this and the product is being removed correctly from the member and are set as `free`